### PR TITLE
Update class args types to be consistent between init and docstring

### PR DESCRIPTION
**Pull request by rnetser**
## Summary by CodeRabbit

- **New Features**
  - Enhanced type handling and clarity for various API components, improving maintainability and type safety.

- **Bug Fixes**
  - Corrected minor documentation issues, such as changing "depreciated" to "deprecated."

- **Documentation**
  - Improved documentation for multiple classes and methods, providing clearer guidance on parameters and their usage.

- **Refactor**
  - Streamlined type mappings and control flow for better readability and maintainability.

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: class_generator/class_generator.py [hunk 1]
```diff
@@ -27,13 +27,43 @@
...
 FIELDS_STR: str = "FIELDS"
 
-TYPE_MAPPING: Dict[str, str] = {
-    "<integer>": "int",
-    "<Object>": "Dict[Any, Any]",
-    "<[]Object>": "List[Any]",
-    "<string>": "str",
-    "<map[string]string>": "Dict[Any, Any]",
-    "<boolean>": "bool",
+PYTHON_TYPE_MAPPING: Dict[Any, str] = {
```
Comment (myakove) [inline class_generator/class_generator.py:30]: Should be `Dict[str, str]`

File: class_generator/tests/manifests/api_server/api_server_res.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class APIServer(Resource):
 
     def __init__(
         self,
-        additional_cors_allowed_origins: Optional[Dict[str, Any]] = None,
```
Comment (myakove) [inline class_generator/tests/manifests/api_server/api_server_res.py:21]: how this changed from Dict to List?

File: class_generator/class_generator.py [hunk 3]
```diff
@@ -337,31 +367,14 @@ def get_arg_params(
...
-
-    if type_from_dict == "int":
-        type_from_dict_for_init = "Optional[int] = None"
+    type_from_dict: Dict[str, str] = TYPE_MAPPING.get(type_, TYPE_MAPPING["<Object>"])
 
     _res: Dict[str, Any] = {
         "name-from-explain": _orig_name,
         "name-for-class-arg": name,
-        "type-for-class-arg": f"{name}: {type_from_dict_for_init}",
+        "type-for-class-arg": f"{name}: {type_from_dict['type_for_init']}",
```
Comment (myakove) [inline class_generator/class_generator.py:375]: why not just get it from the old dict that we had?

File: class_generator/tests/manifests/api_server/api_server_res.py (hunk 2 above)
Comment (rnetser) [reply class_generator/tests/manifests/api_server/api_server_res.py:21]: it is a list (defined as `<[]string>`, before we defaulted to dict if we did not have it. I added this as another type

File: class_generator/class_generator.py (hunk 3 above)
Comment (rnetser) [reply class_generator/class_generator.py:375]: now we have one dict which holds both type and type_for_init; i think it is easier to follow

File: class_generator/class_generator.py [hunk 4]
```diff
@@ -331,37 +322,60 @@ def get_arg_params(
...
+    python_type_mapping: Dict[str, str] = {
+        "int": "int",
+        "dict": "Dict[str, Any]",
+        "str": "str",
+        "bool": "bool",
+        "list_any": "List[Any]",
+        "list_str": "List[str]",
+    }
+    # All fields must be set with Optional since resource can have yaml_file to cover all args.
+    type_mapping: Dict[str, Dict[str, str]] = {
```
Comment (myakove) [inline class_generator/class_generator.py:334]: we should do 
```
if type_ == "<integer>":
    type_for_doc =  "int",
    type_for_init =  f"Optional[int] = None",
```
and so on...
No need any of the dicts

**Review by myakove — CHANGES_REQUESTED** [review]

File: class_generator/class_generator.py (hunk 4 above)
Comment (rnetser) [reply class_generator/class_generator.py:334]: but we have duplications; e.g see `dict`

Comment (myakove) [reply class_generator/class_generator.py:334]: duplicate how?

Comment (myakove) [reply class_generator/class_generator.py:334]: you mean the word `int` for example?

then do 

```
if type_ == "<integer>":
    type_for_doc =  "int",
    type_for_init =  f"Optional[{type_for_doc}] = None",
```

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**